sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLListener.scala

```diff
@@ -113,7 +116,7 @@ class SQLListener(conf: SparkConf) extends SparkListener with Logging {
    */
   private val _jobIdToExecutionId = mutable.HashMap[Long, Long]()
 
-  private val _stageIdToStageMetrics = mutable.HashMap[Long, SQLStageMetrics]()
+  private val _stageIdToStageMetrics = mutable.LinkedHashMap[Long, SQLStageMetrics]()
```

Comment:
Maybe we can use Java's `LinkedHashMap` and override `removeEldestEntry` to what we want.

Reply:
`removeEldestEntry` is a protected method.

Reply:
Java's `LinkedHashMap` can be overridden with an custom implementation of `removeEldestEntry`, that will save the codes done below. It is not the user who call this `removeEldestEntry`...